src/lib/Microsoft.Health.Fhir.Ingest/Service/ResourceIdentityServiceAttribute.cs

```diff
@@ -0,0 +1,24 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Microsoft.Health.Fhir.Ingest.Config;
+
+namespace Microsoft.Health.Fhir.Ingest.Host
+{
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true, Inherited = false)]
```

Comment:
I'm curious if there is a specific reason you are using an assembly attribute rather than a class attribute. If a class attribute was used, you could remove the ClassType property here, and when you create the identity service registry in the ResourceIdentityServiceFactory, you can call GetTypes() on the assemblies and .GetCustomAttributes<ResourceIdentityServiceAttribute>() to find classes that have been attributed.

I'm 100% sure this works as is, I'm just curious if there are other reasons.

Reply:
This is my same concern as well.

Reply:
Good question and thanks for asking also. I did spend some time on evaluating this. So I've replied Dustin's comment with more details. The basic idea is that the class attribute seems requiring me to iterate all types in all assemblies, thinking of a tree, that's all leaf nodes. Rather than that, I'd like to prune some branch out at assembly level. And with the types just added in the attribute, I don't have to go through all types but looking them up.  Please let me know if there is anything missed though.

Reply:
Thanks for the detailed write up. I have 2 opinions when it comes to this:

1. If we decide to use assembly attributes, I think they should be added to AssemblyInfo.cs rather than in each class. To me, it looks out of place to have the assembly attributes in class files.
2. While I'm okay with either solution, my personal preference would be to attribute the class. I realize this is not as efficient, but I think it makes it easier to read and forces the attribute to be written in the class file.

Reply:
I think it makes sense to add them all in the AssemblyInfo.cs, so we can manage them altogether as a registry, which is easier to browse and maintain from another perspective. I'll update once this option is finalized.